docs/book/tutorials/kubernetes-on-vsphere-with-kubeadm.md

```diff
@@ -1,58 +1,137 @@
+kubectl - brew install kubernetes-cli
+tmux (optional) - brew install tmux
+```
 
-## Setting up VMs with K8s components
+Here are the links to the tools and install instructions for other operating systems:
 
-### On all nodes
+* govc for other Operating Systems - version 0.20.0 or higher recommended
+https://github.com/vmware/govmomi/tree/master/govc
+* kubectl for other Operating Systems - https://kubernetes.io/docs/tasks/tools/install-kubectl/
+* tmux for other Operating Systems - https://github.com/tmux/tmux
 
-Install the container runtime (in our case Docker)
+## Setting up VMs and Guest OS
+The next step is to install the necessary Kubernetes components on the Ubuntu OS virtual machines. Some components must be installed on all of the nodes. In other cases, some of the components need only be installed on the master, and in other cases, only the workers. In each case, where the components are installed is highlighted.
+All installation and configuration commands should be executed with root privilege. You can switch to the root environment using the "sudo su" command.
+Setup steps required on all nodes
+The following section details the steps that are needed on both the master and worker nodes.
+### disk.EnableUUID=1
+The following govc commands will set the disk.EnableUUID=1 on all nodes.
+```
+# export GOVC_INSECURE=1
+# export GOVC_URL='https://<VC_Admin_User>:<VC_Admin_Passwd>@<VC_IP>'
+
+# govc ls
+/datacenter/vm
+/datacenter/network
+/datacenter/host
+/datacenter/datastore
+ 
+
+To retrieve all Node VMs, use the following command:
+
+# govc ls /<datacenter-name>/vm
+/datacenter/vm/k8s-node3
+/datacenter/vm/k8s-node4
+/datacenter/vm/k8s-node1
+/datacenter/vm/k8s-node2
+/datacenter/vm/k8s-master
+ 
+
+To use govc to enable Disk UUID, use the following command:
+
+# govc vm.change -vm '/datacenter/vm/k8s-node1' -e="disk.enableUUID=1"
+# govc vm.change -vm '/datacenter/vm/k8s-node2' -e="disk.enableUUID=1"
+# govc vm.change -vm '/datacenter/vm/k8s-node3' -e="disk.enableUUID=1"
+# govc vm.change -vm '/datacenter/vm/k8s-node4' -e="disk.enableUUID=1"
+# govc vm.change -vm '/datacenter/vm/k8s-master' -e="disk.enableUUID=1"
+Upgrade Virtual Machine Hardware 
+VM Hardware should be at version 15 or higher.
+ 
+# govc vm.upgrade -version=15 -vm '/datacenter/vm/k8s-node1'
+# govc vm.upgrade -version=15 -vm '/datacenter/vm/k8s-node2'
+# govc vm.upgrade -version=15 -vm '/datacenter/vm/k8s-node3'
+# govc vm.upgrade -version=15 -vm '/datacenter/vm/k8s-node4'
+# govc vm.upgrade -version=15 -vm '/datacenter/vm/k8s-master'
+```
+### Disable Swap
+SSH into all K8s worker nodes and disable swap on all nodes including master node. This is a prerequisite for kubeadm.
+```
+# swapoff -a
+# vi /etc/fstab # remove any swap entry from this file.
+```
+### Install Docker CE
+The following steps should be used to install the container runtime on all of the nodes. At a minimum, Docker CE 18.06 should be used.
```

Comment:
Not at a minimum - K8s has explicit supported versions, so it _has_ to be this version.

Reply:
Done